aten/src/ATen/native/Itertools.cpp

```diff
@@ -12,7 +12,7 @@ Tensor _triu_mask(int64_t n, int64_t dims, bool diagonal, TensorOptions opt) {
   // or i <= j <= k <= ... (depending on diagonal)
   Tensor range = at::arange(n, opt.dtype(kLong));
   std::vector<Tensor> index_grids = at::meshgrid(std::vector<Tensor>(dims, range));
-  Tensor mask = at::ones(index_grids[0].sizes(), opt.dtype(kByte));
+  Tensor mask = at::ones(index_grids[0].sizes(), opt.dtype(kBool));
```

Comment:
I wonder (you don't have to make this change): does `at::full(..., True, opt.dtype(kBool));` work?  That way you avoid specifying ones for bool.

Reply:
i cant replace at::ones with at::full as first will return [ Variable[CPUBoolType]{SIZE} ] and second will give [ CPUDoubleType{SIZE} ]

Reply:
I think this is because you still need to pass dtype bool.